Cython/Utility/Exceptions.c

```diff
@@ -600,10 +600,8 @@ static void __Pyx_WriteUnraisable(const char *name, CYTHON_UNUSED int clineno,
     PyGILState_STATE state;
     if (nogil)
         state = PyGILState_Ensure();
-#ifdef _MSC_VER
     /* arbitrary, to suppress warning */
     else state = (PyGILState_STATE)-1;
```

Comment:
@da-woods This initialization value triggers the following GCC warning if enabled:

```
#### 2022-10-17 12:57:39.471569
Compiler output for module buffer:
buffer.cpp: In function ‘void __Pyx_WriteUnraisable(const char*, int, int, const char*, int, int)’:
buffer.cpp:7541:36: warning: the result of the conversion is unspecified because ‘-1’ is outside the range of type ‘PyGILState_STATE’ [-Wconversion]
 7541 |     else state = (PyGILState_STATE)-1;
      |                                    ^~
```

Should we change the init value to let say `(PyGILState_STATE)0` (which corresponds to the `PyGILState_LOCKED` enum value) ?

Reply:
Yes that seems reasonable.

Reply:
Fixed.